cli/tfsource/types.go

```diff
@@ -43,9 +45,23 @@ func (src *TerraformSource) String() string {
 // so the same file path (/foo/bar) is always considered the same version. See also the encodeSourceName and
 // ProcessTerraformSource methods.
 func (terraformSource TerraformSource) EncodeSourceVersion() string {
+	if IsLocalSource(terraformSource.CanonicalSourceURL) {
+		hash, err := terraformSource.HashLocalDir()
+		fmt.Printf("\nLocal info!\n\tLocal hash: %s\n\tCanonical URL: %s\n\tCanonical path: %s\n\tWorking Dir: %s\n", hash, terraformSource.CanonicalSourceURL, terraformSource.CanonicalSourceURL.Path, terraformSource.WorkingDir)
+		if err == nil {
+			return hash
+		}
+	}
 	return util.EncodeBase64Sha1(terraformSource.CanonicalSourceURL.Query().Encode())
 }
 
+func (terraformSource TerraformSource) HashLocalDir() (string, error) {
+	if !IsLocalSource(terraformSource.CanonicalSourceURL) {
+		return "", fmt.Errorf("is not a local directory")
```

Comment:
Maybe you could provide some suggestion how to handle this case. This is are kinda my first steps with Go :D
In case that the tfsource is not a local source how should this be handled? Or should the func not be part of the object?

Reply:
Not sure if we need on structure `TerraformSource` function `HashLocalDir` if it is used inside of the package - can be private inside of package

Reply:
Merged into the `EncodeSourceVersion` function.